### PR TITLE
Docs update

### PR DESCRIPTION
**Pull request by stevenmcdonald**
This is a first pass at some docs cleanup, feedback welcome

* A first pass at making the root README a little more friendly to new visitors
* Document the supported protocols and URL formats (most importantly the details on Shadowsocks URLs)
* Fix a couple typos

It might be easier to look rendered as HTML:
https://github.com/greatfire/envoy/blob/docs-update/README.md
https://github.com/greatfire/envoy/blob/docs-update/android/README.md

### CODE REVIEW

**Comment by vitriolix** [comment]
not to say you need to do this in this pr, but note for future we should update wiki unblocked's docs:

https://github.com/greatfire/envoy/blob/docs-update/apps/wikipedia.md

File: android/README.md [hunk 1]
```diff
@@ -22,35 +22,61 @@ CronetNetworking.initializeCronetEngine(getApplicationContext(), "YOUR-ENVOY-URL
...
 ## Envoy url format
 
-Http/Https:
- - http://domain:port (no trailing slash)
- - https://domain:port (no trailing slash)
+### HTTP/HTTPS
 
-Shadowsocks:
- - ss://encrypted login@ip:port/
+Envoy uses a nonstandard HTTP/HTTPS proxy where the origial request is passed in a header. Unencrypted HTTP is supported, but not recommeded. Cronet currently does not support proxying over QUIC / HTTP/3, so HTTP 1.1 or 2 is required. Server side setup is documented [here](./native/README.md) and a working example can be found [here](https://gitlab.com/stevenmcdonald/envoy-proxy-examples/-/tree/main/http_proxy).
```
Comment (vitriolix) [inline android/README.md:27]: ./native/... produces this url:

https://github.com/greatfire/envoy/blob/docs-update/android/native/README.md

i think you want ../native/

File: README.md [hunk 2]
```diff
@@ -11,6 +11,20 @@ Technical details are explained [here](native/README.md).
 
 Please note that this project is unrelated to the [Envoy Proxy project](https://www.envoyproxy.io/).
 
+# Getting Started
+
+## Example Applications
+
+* Wiki Unblocked: a fork of the official Wikipedia app with Envoy support [Play Store](https://play.google.com/store/apps/details?id=org.greatfire.wikiunblocked) [F-Droid](https://f-droid.org/en/packages/org.greatfire.wikiunblocked.fdroid/) [Source](https://github.com/greatfire/apps-android-wikipedia-envoy)
```
Comment (mnbogner) [inline README.md:18]: i've been working off of this branch on wiki unblocked: https://github.com/greatfire/apps-android-wikipedia-envoy/tree/wiki-unblocked

File: android/README.md [hunk 3]
```diff
@@ -22,35 +22,61 @@ CronetNetworking.initializeCronetEngine(getApplicationContext(), "YOUR-ENVOY-URL
...
+
+### [V2Ray](https://github.com/v2fly/v2ray-core)
+
+> Project V is a set of network tools that helps you to build your own computer network. It secures your network connections and thus protects your privacy.
+
+Envoy currently uses a V2Ray fork maintained by a group called [V2Fly](https://www.v2fly.org/en_US/)
+
+Envoy supports two protocols from V2Ray, both based on QUIC. One also masquerades as a WeChat video call, the other masquerades as an SRTP call. V2Ray uses shared knowledge of an UUID for authentication. For this example, we'll use "9e16552c-5de9-4369-95da-db712d7281ee"
+
+If we have a server on 192.168.64.19 with V2Ray 16285, the URLs would be:
```
Comment (mnbogner) [inline android/README.md:76]: should there be something between "V2Ray" and "16285"?

**Comment by mnbogner** [comment]
since you're already updating the readme, do you want to update the "Envoy broadcasts" section to reflect the recent changes?  i could provide the text if you want.

**Review by vitriolix — APPROVED** [review]
looks good to me

**Comment by stevenmcdonald** [comment]
> since you're already updating the readme, do you want to update the "Envoy broadcasts" section to reflect the recent changes? i could provide the text if you want.

Yeah, I'm only vaguely aware of how that works, you should probably write it

**Comment by stevenmcdonald** [comment]
Updated to address all the feedback (other than the Envoy broadcasts :) )

**Comment by mnbogner** [comment]
since you're already updating the readme, do you want to update the "Envoy broadcasts" section to reflect the recent changes?  i could provide the text if you want.

> > since you're already updating the readme, do you want to update the "Envoy broadcasts" section to reflect the recent changes? i could provide the text if you want.
> 
> Yeah, I'm only vaguely aware of how that works, you should probably write it

I pushed some changes to the broadcast section to this branch.

**Pull request merged**